File: VB.NET/Source/Virtusales.Biblio.API.BiblioAPIBase.vb
Namespace Virtusales.Biblio.API
    Public MustInherit Class [BiblioAPIBase]
        Public Property DebugToConsole as Boolean
        
        Sub Debug(Text as String)
            If DebugToConsole Then Console.WriteLine(Text)
        End Sub
    End Class
End Namespace

File: C-Sharp/Source/Virtusales.Biblio.API.BiblioAPIBase.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Virtusales.Biblio.API
{
	public abstract class BiblioAPIBase
	{
		public bool DebugToConsole { get; set; }

		public void Debug(string Text)
		{
			if (DebugToConsole)
				Console.WriteLine(Text);
		}
	}
}